packages/examples/cvat/recording-oracle/src/services/cloud/types.py

```diff
@@ -131,41 +135,44 @@ def from_dict(cls, data: Dict) -> BucketAccessInfo:
         ):
             if required_field not in data:
                 raise ValueError(
-                    f"The {required_field} is required and is not specified in the bucket configuration"
+                    f"The {required_field} is required and is not "
+                    "specified in the bucket configuration"
                 )
 
-        data["provider"] = CloudProvider.from_str(data["provider"].lower())
+        data["provider"] = CloudProviders.from_str(data["provider"].lower())
 
         if (access_key := data.pop("access_key", None)) and (
             secret_key := data.pop("secret_key", None)
         ):
             data["credentials"] = S3BucketCredentials(access_key, secret_key)
 
         elif service_account_key := data.pop("service_account_key", None):
-            data["credentials"] = GCSCredentials(service_account_key)
+            data["credentials"] = GcsBucketCredentials(service_account_key)
 
         return BucketAccessInfo(**data)
 
     @classmethod
-    def from_storage_config(cls, config: Type[IStorageConfig]) -> BucketAccessInfo:
+    def from_storage_config(cls, config: Type[StorageConfig]) -> BucketAccessInfo:
         credentials = BucketCredentials.from_storage_config(config)
 
         return BucketAccessInfo(
-            provider=CloudProvider.from_str(config.provider),
+            provider=CloudProviders.from_str(config.provider),
             host_url=config.provider_endpoint_url(),
             bucket_name=config.data_bucket_name,
             credentials=credentials,
         )
 
     @classmethod
     def parse_obj(
-        cls, data: Union[Dict, str, Type[IStorageConfig]]
+        cls, data: Union[Dict, str, Type[StorageConfig], manifest.BucketUrl]
     ) -> BucketAccessInfo:
-        if isinstance(data, Dict):
+        if isinstance(data, dict):
             return cls.from_dict(data)
+        elif isinstance(data, manifest.BucketUrlBase):
+            return cls.from_dict(data.dict())
         elif isinstance(data, str):
             return cls.from_url(data)
-        elif issubclass(data, IStorageConfig):
+        elif issubclass(data, StorageConfig):
```

Comment:
How about `ExchangeOracleStorageConfig`? it is not a subclass of recording oracle `StorageConfig`

Reply:
I overlooked the interface added, thanks. I think, we can use it, but these configs are not really supposed to share any fields.